### PR TITLE
Add headers to crossgen, cee_crossgen, corzap_crossgen vcxproj files

### PR DESCRIPTION
**Pull request by 4creators**
Work towards #14884

### CODE REVIEW

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT x64 Formatting
@dotnet-bot test Windows_NT x64 Release CoreFX Tests

**Comment by 4creators** [comment]
@janvorli @jkotas CI is green. PTAL

File: src/zap/crossgen/CMakeLists.txt [hunk 1]
```diff
@@ -17,15 +17,42 @@ set(ZAP_SOURCES
...
+    ../zapperstats.h
+    ../nativeformatwriter.h
+)
+
 if (FEATURE_READYTORUN)
     list(APPEND ZAP_SOURCES
         ../zapreadytorun.cpp
         ../nativeformatwriter.cpp
     )
+    list(APPEND ZAP_SOURCES
```
Comment (janvorli) [inline src/zap/crossgen/CMakeLists.txt]: this should be ZAP_HEADERS

Comment (4creators) [reply src/zap/crossgen/CMakeLists.txt]: Ooops, thanks for catching this error.

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT x64 Formatting

**Comment by 4creators** [comment]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_debug_windows_nt_innerloop_prtest/1593/
00:17:44.181 FATAL: command execution failed
00:17:44.183 java.nio.channels.ClosedChannelException

**Comment by 4creators** [comment]
@janvorli Addressed PR feedback. Don't expect that all tests will pass due to known CI problems. PTAL

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT arm Cross Debug Innerloop Build

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT x64 Formatting
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x64 min_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness 
@dotnet-bot test Windows_NT x86 min_opt ryujit CoreCLR Perf Tests Correctness

**Comment by 4creators** [comment]
@dotnet-bot test Windows_NT arm Cross Debug Innerloop Build

**Comment by 4creators** [comment]
@janvorli Ping. Addressed code review feedback. CI is green finally 👍 PTAL

**Comment by 4creators** [comment]
@jankotas @janvorli Ping - PTAL

**Pull request merged**